src/libraries/System.Text.RegularExpressions/tests/Regex.KnownPattern.Tests.cs

```diff
@@ -1405,5 +1407,58 @@ public void TerminationInNonBacktrackingVsBackTracking(RegexOptions options, int
             // NonBacktracking needs way less than 1s
             Assert.False(re.Match(input).Success);
         }
+
+        //
+        // dotnet/runtime-assets contains a set a regular expressions sourced from
+        // permissively-licensed packages.  Validate Regex behavior with those expressions.
+        //
+
+        [Theory]
+        [InlineData(RegexEngine.Interpreter)]
+        [InlineData(RegexEngine.Compiled)]
+        public void PatternsDataSet_ConstructRegexForAll(RegexEngine engine)
+        {
+            Parallel.ForEach(s_patternsDataSet.Value, r =>
+            {
+                try
+                {
+                    RegexHelpers.GetRegexAsync(engine, r.Pattern, r.Options).GetAwaiter().GetResult();
+                }
+                catch (Exception e) when (RegexHelpers.IsNonBacktracking(engine) && e.Message.Contains("NonBacktracking"))
+                {
+                    // Some tests aren't supported with RegexOptions.NonBacktracking.
+                }
+            });
+        }
+
+        [OuterLoop("Takes many seconds")]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "NonBacktracking isn't supported on .NET Framework")]
+        [Fact]
+        public void PatternsDataSet_ConstructRegexForAll_NonBacktracking() =>
+            PatternsDataSet_ConstructRegexForAll(RegexEngine.NonBacktracking);
+
+        [OuterLoop("Takes minutes to generate and compile thousands of expressions")]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "Source generator isn't supported on .NET Framework")]
+        [Fact]
+        public void PatternsDataSet_ConstructRegexForAll_SourceGenerated() =>
+            PatternsDataSet_ConstructRegexForAll(RegexEngine.SourceGenerated);
+
+        private static Lazy<DataSetExpression[]> s_patternsDataSet = new Lazy<DataSetExpression[]>(() =>
+        {
+            using Stream json = File.OpenRead("Regex_RealWorldPatterns.json");
+            return JsonSerializer.Deserialize<DataSetExpression[]>(json, new JsonSerializerOptions() { ReadCommentHandling = JsonCommentHandling.Skip }).Distinct().ToArray();
+        });
+
+        private sealed class DataSetExpression : IEquatable<DataSetExpression>
+        {
+            public int Count { get; set; }
+            public RegexOptions Options { get; set; }
+            public string Pattern { get; set; }
+
+            public bool Equals(DataSetExpression? other) =>
```

Comment:
where is this Equals getting called?

Reply:
By Distinct()

Reply:
oh yes. are there intentionally dupes in there?

Reply:
A non-trivial number of patterns show up multiple times with distinct options, most commonly one with Compiled and one without.